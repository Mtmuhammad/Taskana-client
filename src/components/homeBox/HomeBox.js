// Home page component for to display individual db info

import React from 'react'
import "./HomeBox.scss";

const HomeBox = ({ icon, type, number, bgColor }) => {
  return (
    <>
      <div className="pe-3 mb-5 col-lg-3 col-md-6">
        <div data-testid="homebox" className="card homebox">
          <div data-testid="homebox-body" className="card-body">
            <div className="d-flex flex-row">
              <div
              data-testid="icon-bg"
                style={{ backgroundColor: `${bgColor}` }}
                className="
              box-icon
              d-flex
              align-items-center
              justify-content-center
              rounded-circle
            "
              >
                <i data-testid="homebox-icon" className={icon}></i>
              </div>
              <div className="ms-4 align-self-center">
                <h3 data-testid="homebox-count" className="mb-2">{number}</h3>
                <h5 data-testid="homebox-type" className="mb-0">{type}</h5>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default HomeBox;
