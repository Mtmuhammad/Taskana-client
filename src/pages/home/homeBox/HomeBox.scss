@use "../../../styles/abstracts/_variables" as *;
@use "../../../styles/abstracts/_mixins" as *;

.box-icon {
   line-height: 6.5rem;
    width: 6rem;
    height: 6rem;
    font-size: 3rem;
}

.homebox {
   background-color: var(--primary-color-light);
   color: var(--text-color);
}



.card-body {
   padding: 3rem;
}