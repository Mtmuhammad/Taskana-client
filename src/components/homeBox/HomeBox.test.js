import React from "react";
import { render, screen } from "@testing-library/react";
import HomeBox from "./HomeBox";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(
      <HomeBox
        bgColor="#00d09c90"
        number="1"
        type="Projects"
        icon="bx bx-briefcase-alt icon"
      />,
      { wrapper: BrowserRouter }
    );
  });

  test("should match snapshot", () => {
    const { asFragment } = render(
      <HomeBox
        bgColor="#00d09c90"
        number="1"
        type="Projects"
        icon="bx bx-briefcase-alt icon"
      />,
      { wrapper: BrowserRouter }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("homebox", () => {
  test("should render homebox", () => {
    render(
      <HomeBox
        bgColor="#00d09c90"
        number="1"
        type="Projects"
        icon="bx bx-briefcase-alt icon"
      />,
      { wrapper: BrowserRouter }
    );
    const homeBox = screen.getByTestId("homebox");
    expect(homeBox).toBeInTheDocument();
    expect(homeBox).toHaveClass("card homebox");
  });
  test("should render homebox body", () => {
    render(
      <HomeBox
        bgColor="#00d09c90"
        number="1"
        type="Projects"
        icon="bx bx-briefcase-alt icon"
      />,
      { wrapper: BrowserRouter }
    );
    const body = screen.getByTestId("homebox-body");
    expect(body).toBeInTheDocument();
    expect(body).toHaveClass("card-body");
  });
  test("should render icon background", () => {
    render(
      <HomeBox
        bgColor="#00d09c90"
        number="1"
        type="Projects"
        icon="bx bx-briefcase-alt icon"
      />,
      { wrapper: BrowserRouter }
    );
    const body = screen.getByTestId("icon-bg");
    expect(body).toBeInTheDocument();
    expect(body).toHaveClass("box-icon rounded-circle");

    const icon = screen.getByTestId("homebox-icon");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass("bx bx-briefcase-alt icon");
  });
  test("should render homebox text", () => {
    render(
      <HomeBox
        bgColor="#00d09c90"
        number="1"
        type="Projects"
        icon="bx bx-briefcase-alt icon"
      />,
      { wrapper: BrowserRouter }
    );
    const count = screen.getByTestId("homebox-count");
    expect(count).toBeInTheDocument();
    const type = screen.getByTestId("homebox-type");
    expect(type).toBeInTheDocument();
  });
});
