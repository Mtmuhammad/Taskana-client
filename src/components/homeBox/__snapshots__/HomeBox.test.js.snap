// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="pe-3 mb-5 col-lg-3 col-md-6"
  >
    <div
      class="card homebox"
      data-testid="homebox"
    >
      <div
        class="card-body"
        data-testid="homebox-body"
      >
        <div
          class="d-flex flex-row"
        >
          <div
            class="
 box-icon
 d-flex
 align-items-center
 justify-content-center
 rounded-circle
 "
            data-testid="icon-bg"
            style="background-color: rgba(0, 208, 156, 0.565);"
          >
            <i
              class="bx bx-briefcase-alt icon"
              data-testid="homebox-icon"
            />
          </div>
          <div
            class="ms-4 align-self-center"
          >
            <h3
              class="mb-2"
              data-testid="homebox-count"
            >
              1
            </h3>
            <h5
              class="mb-0"
              data-testid="homebox-type"
            >
              Projects
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
